first-party/jib-classpath-main-class-extension-gradle/src/main/java/com/google/cloud/tools/jib/gradle/extension/classpathmainclass/ClasspathMainClassExtractor.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.gradle.extension.classpathmainclass;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+public class ClasspathMainClassExtractor {
+
+  private static List<String> classpathOptions = Arrays.asList("-cp", "-classpath", "--class-path");
+
+  static Map<String, String> extractEnvsFromEntrypoint(@Nullable List<String> entrypoint) {
```

Comment:
We can use `@VisibleForTesting`, right?

Reply:
this annotation doesn't seem to do anything: https://guava.dev/releases/19.0/api/docs/com/google/common/annotations/VisibleForTesting.html
also documentation says "Annotates a program element that exists, or is more widely visible than otherwise necessary, only for use in test code." but this method needs package-private access, it doesn't have elevated access only for testing.

Reply:
Yes, the annotation doesn't execute any code; it's for documentation purposes. But I just realized you're right that the method has to be package-private. So we're good.